broker-daemon/utils/migrate-store.js

```diff
@@ -7,20 +7,47 @@
  * @param  {number}   batchSize         - Number of operations to batch before executing on the target database
  * @returns {Promise<void>}
  */
-async function migrateStore (sourceStore, targetStore, createDbOperation, batchSize = 20) {
+async function migrateStore (sourceStore, targetStore, createDbOperation, batchSize = 2000) {
   return new Promise((resolve, reject) => {
     const stream = sourceStore.createReadStream()
+
+    // `batchInProgress` allows us to batch save records in sequence and lock the operation
+    // to make sure we are not inserting records out of the order they are received.
+    //
+    // We set this value to automatically resolve on the first use. This value is then
+    // subsequently set when a batch is processed
+    let batchInProgress = async () => {}
+    let batchResolve
+
     let batch = []
 
-    function flush (done = () => {}) {
+    async function flush (done = () => {}) {
       if (!batch.length) {
         return process.nextTick(done)
       }
 
-      targetStore.batch(batch, (err) => {
-        if (err) return reject(err)
-        done()
-      })
+      await batchInProgress()
+
+      batchInProgress = () => {
+        return new Promise((resolve, reject) => {
+          batchResolve = resolve
+        })
+      }
+
+      try {
+        targetStore.batch(batch, (err) => {
```

Comment:
why not wrap `batch` in `promisify` like we do elsewhere? Then you manipulate the promise directly.